### ISSUE TITLE
Consistent use of underscores

### ISSUE THREAD

**Issue by larsmans** [NLeSC/xtas#111]
As noted at #110 and #105, underscores should be consistently applied to mark public/private boundaries, and we should have this documented somewhere. We have:

* public API
* private functions (with _)
* private modules (with _), but "semi-public" functions within them that are used in other modules

Ping @LourensVeen. Maybe put this in the contributing code docs?

**Comment by LourensVeen** [comment]
Yes, that would be a good place to put some explanation. I'll look at it, if you want to assign me to this issue go ahead.

**Comment by larsmans** [comment]
Yes, please work on this; I'm refactoring other parts of the code.
